### ISSUE TITLE
hovered line in detail display is wrong

### ISSUE THREAD

**Issue by mattijn** [vega/vega-lite#4603]
First of all, big kudos for releasing rc13 especially the new `init` functionality!

I try to include a hover selection on the detail display for a rule within a detail+overview visualisation. But the rule in the detailed panel seems to follow the domain like if it was not zoomed in.

See gif
![kapture 2019-02-25 at 21 50 47](https://user-images.githubusercontent.com/5186265/53368114-2e604380-3948-11e9-831d-6d0931041dc1.gif)

And spec
```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
  "data": {"url": "data/sp500.csv"},
  "vconcat": [
    {
      "encoding": {
        "x": {
          "field": "date",
          "type": "temporal",
          "scale": {"domain": {"selection": "brush", "encoding": "x"}},
          "axis": {"title": ""}
        },
        "color": {"value":"red"},
        "tooltip": [
          {"timeUnit": "yearmonthdate", "field": "date", "type": "temporal"},
          {"field": "price", "type": "quantitative"}
        ]
      },
      "layer": [
        {
          "width": 480,
          "mark": "line",
          "encoding": {"y": {"field": "price", "type": "quantitative"}}
        },
        {
          "mark": "rule",
          "selection": {
            "hover": {
              "type": "single",
              "on": "mouseover",
              "empty": "none",
              "nearest": true
            }
          },
          "encoding": {
            "opacity": {
              "condition": {"selection": "hover", "value": 1},
              "value": 0
            }
          }
        },
        {
          "mark": {"type": "point", "opacity": 1, "size": 100, "fill": "white"},
          "encoding": {
            "y": {"field": "price", "type": "quantitative"},
            "opacity": {
              "condition": {"selection": "hover", "value": 1},
              "value": 0
            }
          }
        }
      ]
    },
    {
      "width": 480,
      "height": 60,
      "mark": "line",
      "selection": {
        "brush": {
          "type": "interval",
          "encodings": ["x"],
          "init": {"x": ["1136070000000", "1230764400000"]}
        }
      },
      "encoding": {
        "x": {"field": "date", "type": "temporal"},
        "y": {
          "field": "price",
          "type": "quantitative",
          "axis": {"tickCount": 3, "grid": false}
        }
      }
    }
  ]
}
```

**Comment by arvind** [comment]
Thanks @mattijn! This issue should be fixed once #4289 is merged.

**Comment by nicodelatorre7** [comment]
Sorry, I have the same problem, what do I have to do to fix it?

`
{
  config: { view: { continuousWidth: 400, continuousHeight: 300 } },
  data: { name: "data-60e3b6f2e240ab5be59fc377b9a014c5" },
  mark: "bar",
  encoding: {
    color: {
      condition: { value: "green", test: "(datum.nonfarm_change > 0)" },
      value: "red",
    },
    tooltip: [
      { field: "month", title: "Date", type: "temporal" },
      { field: "nonfarm_change", title: "Value", type: "quantitative" },
    ],
    x: { field: "month", type: "temporal" },
    y: { field: "nonfarm_change", type: "quantitative" },
  },
  height: 150,
  width: 600,
  $schema: "https://vega.github.io/schema/vega-lite/v4.17.0.json",
    datasets: {
    "data-60e3b6f2e240ab5be59fc377b9a014c5": [
      {
        month: "2006-01-01",
        nonfarm_change: 282,
        private: 113603,
        ......
        ......
     }
`

Thanks
